lsst_efd_client/efd_helper.py

```diff
@@ -377,9 +381,49 @@ async def select_packed_time_series(self, topic_name, base_fields, start, end,
         result = await self.select_time_series(topic_name, field_list+[ref_timestamp_col, ],
                                                start, end, is_window=is_window, index=index)
         vals = {}
-        for field in base_fields:
-            df = merge_packed_time_series(result, field, ref_timestamp_col=ref_timestamp_col,
+        for f in base_fields:
+            df = merge_packed_time_series(result, f, ref_timestamp_col=ref_timestamp_col,
                                           internal_time_scale=self.internal_scale)
-            vals[field] = df[field]
+            vals[f] = df[f]
         vals.update({'times': df['times']})
         return pd.DataFrame(vals, index=df.index)
+
+    async def get_schema(self, topic):
+        """Givent a topic, get a list of dictionaries describing the fields
+
+        Parameters
+        ----------
+        topic : `str`
+            The name of the topic to query.  A full list of valid topic names
+            can be obtained using ``get_schema_topics``.
+
+        Returns
+        -------
+        result : `Pandas.DataFrame`
+            A dataframe with the schema information for the topic.  One row per field.
+        """
+        async with aiohttp.ClientSession() as http_session:
+            registry_api = RegistryApi(
+                session=http_session, url=self.schema_registry
+            )
+            schema = await registry_api.get_schema_by_subject(f'{topic}-value')
+            return self._parse_schema(topic, schema)
+
+    @staticmethod
+    def _parse_schema(topic, schema):
+        # A helper function so we can test our parsing
+        fields = schema['schema']['__named_schemas'][topic]['fields']
+        vals = {'name': [], 'description': [], 'units': [], 'aunits': []}
+        for f in fields:
+            vals['name'].append(f['name'])
+            vals['description'].append(f['description'])
+            vals['units'].append(f['units'])
+            try:
+                if vals['units'][-1] == 'unitless':  # Special case not having units
+                    vals['aunits'].append(u.dimensionless_unscaled)
+                else:
+                    vals['aunits'].append(u.Unit(vals['units'][-1]))
+            except (ValueError, TypeError) as e:
+                logging.warning(f'Could not construct unist: {e.args[0]}')
+                vals['aunits'].append(None)
+        return pd.DataFrame(vals)
```

Comment:
Looks great! The implementation and the interface look much simpler now with `get_topics()`, `get_fields()` and `get_schema()` all returning a Pandas Dataframe.

It looks like `get_schema()` did not make in to the docs? I'm looking at https://efd-client.lsst.io/v/DM-29100/index.html

Oh, I see. It is there if I navigate to `EfdClient` but it didn't show up in the first page.

Reply:
Also to answer the previous question, the Avro schema is used by the InfluxDB connector to deserialize an Avro record when writing to InfluxDB. So, if the topic exists in InfluxDB it is guaranteed that the corresponding schema exists. The contrary is not, you can have a schema created by SAL Kafka during initialization, but the topic is only created in InfluxDB when the CSC publishes the first record. That's why `SHOW MEASUREMENTS` is more robust in returning the list of topics that actually exist.

Reply:
Thanks for pointing out the missing documentation.  There is an overview section where I manually added descriptions of the useful functions and didn't manage to add that.